src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1750,6 +1751,76 @@ public static void ConsumeLotsOfComments(string valueString, bool insideArray, s
             Assert.Equal(dataUtf8.Length, json.BytesConsumed);
         }
 
+        [Theory]
+        [MemberData(nameof(SingleLineCommentData))]
+        public static void ConsumeSingleLineCommentSingleSpanTest(string[] inputData)
+        {
+            string expected = inputData[inputData.Length - 1];
+            inputData = inputData.Take(inputData.Length - 1).ToArray();
+
+            var state = new JsonReaderState(options: new JsonReaderOptions { CommentHandling = JsonCommentHandling.Allow });
+            int bytesConsumed = 0;
+            var dataUtf8 = new byte[] { };
+
+            for (int count = 0; count < inputData.Length; ++count)
+            {
+                bool isFinal = (count == inputData.Length - 1);
+                dataUtf8 = dataUtf8.Concat(Encoding.ASCII.GetBytes(inputData[count])).ToArray();
+                var json = new Utf8JsonReader(dataUtf8, isFinalBlock: isFinal, state);
+                while (json.Read())
+                {
+                    switch (json.TokenType)
+                    {
+                        case JsonTokenType.StartObject:
+                        case JsonTokenType.EndObject:
+                            break;
+                        case JsonTokenType.Comment:
+                            Assert.Equal(Encoding.ASCII.GetBytes(expected), json.ValueSpan.ToArray());
+                            break;
+                        default:
+                            Assert.True(false);
+                            break;
+                    }
+                }
+                dataUtf8 = dataUtf8.Skip(bytesConsumed).ToArray();
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SingleLineCommentData))]
+        public static void SkipSingleLineCommentSingleSpanTest(string[] inputData)
```

Comment:
Maybe restructure this as `string input, string expected` rather than a string array.

Reply:
In its current form, the test is a bit difficult to follow, especially as you build up `dataUtf8` and re-use `inputData` as both input and expected output.

Reply:
> Maybe restructure this as `string input, string expected` rather than a string array.

> In its current form, the test is a bit difficult to follow, especially as you build up dataUtf8 and re-use inputData as both input and expected output.

Redone the input data to split it into _expected_ string and an array of strings which are fragments of input JSON; one or two fragments generated by splitting the JSON before and after the line separator (or combination thereof). Have added relevant comments to `SingleLineCommentData` method. Note that `SingleLineCommentData` is used in both `Skip...` and `Consume...` test cases.